src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+
+            returned_string = returned_string.decode('utf-8')
+            lines = returned_string.splitlines()
+
+            for line in lines:
+
+                print(line, file=stderr)
+
+                # Skip empty lines
+                if not line.strip():
+                    continue
+
+                # Check if we want to mark the start of a new section
+                # Check this by seeing if the line is a top level description, ie: 'Commands:'
+                # These are characterized by a new line with text starting without spaces
+                read_section_output = self._validate_output_read_section_name(line)
+                if read_section_output is not None:
+                    section = read_section_output
+                    continue
+
+                    # If this line is a section designator, no additional processing
+                    # is required. Move on to the next line.
+
+                updated_indices = self._validate_output_validate_line(command_input, line, section,
+                                                                      subgroups, commands,
+                                                                      subgroups_index,
+                                                                      commands_index)
+                subgroups_index = updated_indices[0]
+                commands_index = updated_indices[1]
+
+            # If section is still 'Start', the something has gone wrong
+            self.assertNotEqual('Start', section,
+                                msg='Command {0} has incomplete help text: {1}'.format(help_command, returned_string)) #pylint: disable=line-too-long
+
+            # Check that we have traversed completely through both
+            # subgroups and commands
+            self.assertEqual(len(commands), commands_index,
+                             msg=('Not all commands listed in help text for '
+                                  + help_command
+                                  + '. This may be a problem due incorrect expected ordering.'))
+            self.assertEqual(len(subgroups), subgroups_index,
+                             msg=('Not all subgroups listed in help text for '
+                                  + help_command
+                                  + '. This may be a problem due incorrect expected ordering.'))
+
+            print(file=stderr)
+
+        except Exception: # pylint: disable=broad-except
+            self.fail(msg='ERROR: Command {0} returned error at execution. Output: {1} Error: {2}'.format(help_command, returned_string, err)) #pylint: disable=line-too-long
+
+        # Once validation is done for the provided command_input,
+        # check if there are any commands returned
+        # in the help text. If so, validate those commands.
+        for command in commands:
+            self.validate_output(command_input + ' ' + command)
+
+    def help_documentation_test(self):
```

Comment:
Use one test per command, not one test for every command

Reply:
What is the reason for this? Thanks!